lib/resize.js

```diff
@@ -438,11 +438,17 @@ function extract (options) {
  * The `info` response Object, obtained from callback of `.toFile()` or `.toBuffer()`,
  * will contain `trimOffsetLeft` and `trimOffsetTop` properties.
  *
- * @param {number} [threshold=10] the allowed difference from the top-left pixel, a number greater than zero.
+ * @param {String|Object} [background='top-left pixel'] background colour, parsed by the [color](https://www.npmjs.org/package/color) module, defaults to that of the top-left pixel.
+ * @param {number} [threshold=10] the allowed difference from the above colour, a number greater than zero.
  * @returns {Sharp}
  * @throws {Error} Invalid parameters
  */
-function trim (threshold) {
+function trim (background, threshold) {
```

Comment:
Perhaps we should make the signature accept an Object of options, so `background` would be specified via:
```js
sharp().trim({ background: 'red' })...
```
so if `threshold` is also required, this would be via:
```js
sharp().trim({ background: 'red', threshold: 20 })...
```
whilst ensuring the existing, number-is-threshold approach still works:
```js
sharp().trim(20)...
```

Reply:
Ah, that looks like a clear solution, thank you!